src/Features/Core/Portable/CodeRefactorings/AddMissingImports/AbstractAddMissingImportsFeatureService.cs

```diff
@@ -157,16 +157,34 @@ private async Task<Document> ApplyFixesAsync(Document document, ImmutableArray<A
             newProject = newProject.AddMetadataReferences(allAddedMetaDataReferences);
             newProject = newProject.AddProjectReferences(allAddedProjectReferences);
 
-            // Only consider insertion changes to reduce the chance of producing a
-            // badly merged final document. Alphabetize the new imports, this will not
-            // change the insertion point but will give a more correct result. The user
-            // may still need to use organize imports afterwards.
-            var orderedTextInserts = allTextChanges.Where(change => change.Span.IsEmpty)
+
+            // The text changes produced by the AddImport code fixes with be inserts except in
+            // the case where there were multiple trailing newlines.
+
+            // Insertion case (zero or one trailing newline following the using directives):
+            // using System;  // Added using directives will all be insertion changes on the following line
+            //
+            // namespace Foo
+            // {
+
+            // Replacement case (two or more trailing newlines following the using directives):
+            // using System; // Added using directives will all be replacement changes that overwrite the following line
```

Comment:
Just curious, why this is done this way before? And why don't we fix it when individual change is created by `IAddImportFeatureService`?

Reply:
Dug into it and during formatting the ElasticTrviaFormattingRule special cases the last import in the list to be followed by 2 line breaks. Since these new imports are given the ElasticTrivia annotation by the SyntaxFactory and formats this way when the new import is added to the end of the list.